folding-schemes/src/folding/hypernova/mod.rs

```diff
@@ -300,10 +484,26 @@ where
         &mut self,
         mut rng: impl RngCore,
         external_inputs: Vec<C1::ScalarField>,
+        other_instances: Self::MultiCommittedInstanceWithWitness,
     ) -> Result<(), Error> {
         // `sponge` is for digest computation.
         let sponge = PoseidonSponge::<C1::ScalarField>::new(&self.poseidon_config);
 
+        #[allow(clippy::type_complexity)]
+        let (lcccs, cccs): (
+            Vec<(LCCCS<C1>, Witness<C1::ScalarField>)>,
+            Vec<(CCCS<C1>, Witness<C1::ScalarField>)>,
+        ) = other_instances;
+
+        let (Us, Ws): (Vec<LCCCS<C1>>, Vec<Witness<C1::ScalarField>>) = lcccs
+            .iter()
+            .map(|inst| (inst.0.clone(), inst.1.clone()))
```

Comment:
It looks that we are cloning those twice, both here and below, when calling `NIMFS::prove`, maybe there could be way around this, such as passing those as references instead?

Reply:
Right, turns out that was easier to directly use
```rust
let (Us, Ws): (Vec<LCCCS<C1>>, Vec<Witness<C1::ScalarField>>) = lcccs.into_iter().unzip();
```
updated to that.